src/actions/field-actions.js

```diff
@@ -14,304 +14,316 @@ import isValid from '../form/is-valid';
+    (model, fieldValidators, options = {}) => (dispatch, getState) => {
+      const value = s.get(getState(), model);
 
-      if (fieldsValidity.hasOwnProperty(fieldKey)) {
-        return true;
-      }
+      const fieldsValidity = mapValues(fieldValidators, (validator, field) => {
+        const fieldValue = field
+          ? s.get(value, field)
+          : value;
 
-      return isValid(form[fieldKey]);
-    });
+        const fieldValidity = getValidity(validator, fieldValue);
 
-  return valid;
-}
+        return fieldValidity;
+      });
 
-const validateFields = trackable((model, fieldValidators, options = {}) => (dispatch, getState) => {
-  const value = _get(getState(), model);
+      const validCB = options.onValid;
+      const invalidCB = options.onInvalid;
 
-  const fieldsValidity = mapValues(fieldValidators, (validator, field) => {
-    const fieldValue = field
-      ? _get(value, field)
-      : value;
+      if (validCB || invalidCB) {
+        const form = getForm(getState(), model);
+        const formValid = (form && !fieldsValidity.hasOwnProperty(''))
+          ? isFormValidWithoutFields(form, fieldsValidity)
+          : true;
 
-    const fieldValidity = getValidity(validator, fieldValue);
+        const fieldsValid = options.errors
+          ? !isValidityInvalid(fieldsValidity)
+          : isValidityValid(fieldsValidity);
 
-    return fieldValidity;
-  });
+        if (validCB && formValid && fieldsValid) {
+          validCB();
+        } else if (invalidCB) {
+          invalidCB();
+        }
+      }
 
-  const validCB = options.onValid;
-  const invalidCB = options.onInvalid;
+      const fieldsValiditySetter = options.errors
+        ? setFieldsErrors
+        : setFieldsValidity;
 
-  if (validCB || invalidCB) {
-    const form = getForm(getState(), model);
-    const formValid = (form && !fieldsValidity.hasOwnProperty(''))
-      ? isFormValidWithoutFields(form, fieldsValidity)
-      : true;
+      dispatch(fieldsValiditySetter(model, fieldsValidity));
+    };
 
-    const fieldsValid = options.errors
-      ? !isValidityInvalid(fieldsValidity)
-      : isValidityValid(fieldsValidity);
+  const validateFieldsErrors = (model, fieldErrorsValidators, options = {}) =>
+    validateFields(model, fieldErrorsValidators, {
+      ...options,
+      errors: true,
+    });
 
-    if (validCB && formValid && fieldsValid) {
-      validCB();
-    } else if (invalidCB) {
-      invalidCB();
-    }
-  }
+  return mapValues({
+    blur,
+    focus,
+    submit,
+    submitFields,
+    validSubmit,
+    setDirty,
+    setErrors,
+    setInitial,
+    setPending,
+    setValidating,
+    setPristine,
+    setSubmitted,
+    setSubmitFailed,
+    setTouched,
+    setUntouched,
+    setValidity,
+    setFieldsValidity,
+    setFieldsErrors,
+    resetValidity,
+    resetErrors,
+    validate,
+    validateErrors,
+    validateFields,
+    validateFieldsErrors,
+    asyncSetValidity,
+  }, trackable);
+}
```

Comment:
You did!  I copied the idea over from [model-actions.js](https://github.com/davidkpiano/react-redux-form/commit/d740f34f7e528eede178d3a7b63e92b06b27b6a5). ;)